src/socket/udp_socket.cr

```diff
@@ -88,4 +88,139 @@ class UDPSocket < IPSocket
     bytes_read, sockaddr, addrlen = recvfrom(message)
     {bytes_read, IPAddress.from(sockaddr, addrlen)}
   end
+
+  # Reports whether transmitted multicast packets should be copied and sent
+  # back to the originator.
+  def multicast_loopback?
+    case @family
+    when Family::INET
+      optname = LibC::IP_MULTICAST_LOOP
+      level = LibC::IPPROTO_IP
+    when Family::INET6
+      optname = LibC::IPV6_MULTICAST_LOOP
+      level = LibC::IPPROTO_IPV6
+    else
+      raise "Unsupported IP address family: #{family}"
```

Comment:
This should raise a more specific `Socket::Error`. DItto for all other generic `raise` calls.

Reply:
Also it would be nice if you'd consistently use either `family` or `@family`.